ocaml/testsuite/tests/capsule-api/poisoning.ml

```diff
@@ -0,0 +1,96 @@
+(* TEST
+ include stdlib_alpha;
+ flags = "-extension-universe alpha";
+ runtime5;
+ { bytecode; }
+ { native; }
+*)
+
+module Capsule = Stdlib_alpha.Capsule
+
+let m = Capsule.create_with_mutex ()
```

Comment:
Could you elaborate? I think this API is sufficient for Stdlib, and we can build on it to expose something like `'a Guarded.t` to users.

Reply:
I mean something like:
```ocaml
module Create_with_mutex () : sig
  type k
  val mutex : k Mutex.t
end
```
which you can then use like:
```ocaml
module M = Capsule.Create_with_mutex()
let m = M.mutex
```
If you want a global mutex protecting some global data then you probably want to do it this way.

It can technically be built from the existing `create_with_mutex` via:
```ocaml
module type S = sig type k val mutex : k Mutex.t end

let create_with_mutex_module () =
  let (P (type k) (m : k Mutex.t)) = create_with_mutex () in
  (module struct type nonrec k = k let mutex = m end : S)
  
module Create_with_mutex () = struct
  include (val create_with_mutex_module ())
end
```
so we could indeed leave it out of the Stdlib if we want, but it might be better to just define it directly here.

Reply:
I think it's best to leave this to libraries. Users shouldn't directly access `Stdlib` anyway.